src/gt4py/next/iterator/transforms/pass_manager.py

```diff
@@ -157,5 +161,8 @@ def apply_fieldview_transforms(
     ir = inline_fundefs.prune_unreferenced_fundefs(ir)
     ir = InlineLambdas.apply(ir, opcount_preserving=True, force_inline_lambda_args=True)
     ir = CollapseTuple.apply(ir, offset_provider=offset_provider)  # type: ignore[assignment] # type is still `itir.Program`
+    ir = inline_dynamic_shifts.InlineDynamicShifts.apply(
+        ir
+    )  # domain inference does not support dynamic offsets yet
     ir = infer_domain.infer_program(ir, offset_provider=offset_provider)
```

Comment:
Would it make sense to support unknown domain in the type inference for `as_offset`?

Reply:
I think this is a useful option for the domain inference and it naturally emerged after I addressed your review comments so I added it. I do however prefer working with an IR that has all domains inferred, which is the case after `InlineDynamicShifts` so I would leave it like that for now.